lib/src/rules/always_require_non_null_named_parameters.dart

```diff
@@ -68,50 +68,64 @@ class _Visitor extends SimpleAstVisitor<void> {
 
   @override
   void visitFormalParameterList(FormalParameterList node) {
-    final params = node.parameters
-        // only named parameters
-        .where((p) => p.isNamed)
-        .map((p) => p as DefaultFormalParameter)
-        // without default value
-        .where((p) => p.defaultValue == null)
-        // without @required
-        .where((p) => !p.declaredElement.hasRequired)
-        .toList();
+    List<DefaultFormalParameter> getParams() {
+      final params = <DefaultFormalParameter>[];
+      for (final p in node.parameters) {
+        // Only named parameters
+        if (p.isNamed) {
+          DefaultFormalParameter parameter = p as DefaultFormalParameter;
+          // Without a default value or marked @required
+          if (parameter.defaultValue == null &&
+              !parameter.declaredElement.hasRequired) {
+            params.add(parameter);
+          }
+        }
+      }
+      return params;
+    }
+
     final parent = node.parent;
     if (parent is FunctionExpression) {
-      _checkParams(params, parent.body);
+      _checkParams(getParams(), parent.body);
     } else if (parent is ConstructorDeclaration) {
-      _checkInitializerList(params, parent.initializers);
-      _checkParams(params, parent.body);
+      _checkInitializerList(getParams(), parent.initializers);
+      _checkParams(getParams(), parent.body);
     } else if (parent is MethodDeclaration) {
-      _checkParams(params, parent.body);
+      _checkParams(getParams(), parent.body);
+    }
+  }
+
+  void _checkAsserts(
+      List<Expression> asserts, List<DefaultFormalParameter> params) {
```

Comment:
Consider converting this to check a single assert condition rather than a list of conditions. Invocation sites can then invoke this method directly rather than creating a list that needs to be iterated over.

Reply:
One less list allocation.  Great idea.  Thanks!